### PR TITLE
[DESK-180] Add option to mute via the api

### PR DESCRIPTION
**Pull request by SilverBirchh**
We want to be able to tell matrix to be mute/unmute a thread

### CODE REVIEW

File: src/PlatformAPI.ts [hunk 1]
```diff
@@ -197,6 +197,7 @@ export interface PlatformAPI {
 
   markAsUnread?: (threadID: ThreadID, messageID?: MessageID) => Awaitable<void>
   archiveThread?: (threadID: ThreadID, archived: boolean) => Awaitable<void>
+  muteThread?: (threadID: ThreadID, muted: boolean) => Awaitable<void>
```
Comment (KishanBagaria) [inline src/PlatformAPI.ts:200]: rn the sdk supports this via `updateThread`, some examples:

Screenshot 2024-06-18 at 9 37 49 PM

File: src/enums.ts [hunk 2]
```diff
@@ -80,6 +80,7 @@ export enum Attribute {
 
   CAN_FETCH_LINK_PREVIEW = 'can_fetch_link_preview',
   CAN_REMOVE_LINK_PREVIEW = 'can_remove_link_preview',
+  SUPPORTS_MUTING = 'supports_muting',
```
Comment (KishanBagaria) [inline src/enums.ts:83]: this is good. i wonder if we can assume all platforms support it

File: src/PlatformAPI.ts (hunk 1 above)
Comment (SilverBirchh) [reply src/PlatformAPI.ts:200]: Missed this! Thank you, I'll just use this by the looks of it

Comment (SilverBirchh) [reply src/PlatformAPI.ts:200]: Do i use a similar method to tell the Texts that the native app muted the thread?

Comment (KishanBagaria) [reply src/PlatformAPI.ts:200]: yep that can be used with state sync update event

**Pull request closed without merging**